apache2/apache2_config.c

```diff
@@ -413,10 +413,14 @@ void *merge_directory_configs(apr_pool_t *mp, void *_parent, void *_child)
             ap_log_perror(APLOG_MARK, APLOG_STARTUP|APLOG_NOERRNO, 0, mp, "Using parent rules in this context.");
             #endif
 
-            /* Copy the rules from the parent context. */
-            merged->ruleset = msre_ruleset_create(parent->ruleset->engine, mp);
             /* TODO: copy_rules return code should be taken into consideration. */
-            copy_rules(mp, parent->ruleset, merged->ruleset, child->rule_exceptions);
+            if(child->rule_exceptions->nelts > 0) {
```

Comment:
Can you describe (perhaps put in a comment) when this situation happens?

Reply:
child->rule_exceptions->nelts > 0 only when we will remove rules at the runtime (ctl:ruleRemoveById, ctl:ruleRemoveByMsg, ctl:ruleRemoveByTag)